File: scripts/extract_pdf_text.py
#!/usr/bin/env python3
import os
import PyPDF2

def extract_pdf_text(pdf_path):
    """Extract text from a PDF file"""
    text = ""
    try:
        with open(pdf_path, 'rb') as file:
            pdf_reader = PyPDF2.PdfReader(file)
            num_pages = len(pdf_reader.pages)
            print(f"  Total pages: {num_pages}")

            for page_num in range(num_pages):
                page = pdf_reader.pages[page_num]
                text += f"\n--- Page {page_num + 1} ---\n"
                text += page.extract_text()

                if (page_num + 1) % 10 == 0:
                    print(f"  Processed {page_num + 1}/{num_pages} pages...")

    except Exception as e:
        print(f"  Error: {e}")
        return None

    return text

def main():
    resources_dir = os.path.join(os.path.dirname(__file__), '../SonoPassresources')
    pdf_files = [f for f in os.listdir(resources_dir) if f.endswith('.pdf')]

    print(f"Found {len(pdf_files)} PDF files:\n")

    for pdf_file in pdf_files:
        print(f"{'='*80}")
        print(f"Processing: {pdf_file}")
        print('='*80)

        pdf_path = os.path.join(resources_dir, pdf_file)
        text = extract_pdf_text(pdf_path)

        if text:
            output_file = os.path.join(
                os.path.dirname(__file__),
                f"{os.path.splitext(pdf_file)[0]}.txt"
            )

            with open(output_file, 'w', encoding='utf-8') as f:
                f.write(text)

            print(f"✓ Extracted text saved to: {output_file}")
            print(f"  Text length: {len(text)} characters")
            print(f"  First 1000 characters:\n{text[:1000]}\n")
        else:
            print(f"✗ Failed to extract text from {pdf_file}\n")

if __name__ == '__main__':
    main()

File: scripts/extract-pdf-text.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { PdfReader } from 'pdfreader';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function extractPDFText(pdfPath) {
  return new Promise((resolve, reject) => {
    let text = '';
    let currentPage = 0;

    new PdfReader().parseFileItems(pdfPath, (err, item) => {
      if (err) {
        reject(err);
      } else if (!item) {
        // End of file
        resolve(text);
      } else if (item.page) {
        // New page
        currentPage = item.page;
        text += `\n--- Page ${currentPage} ---\n`;
      } else if (item.text) {
        // Text item
        text += item.text + ' ';
      }
    });
  });
}

async function main() {
  const resourcesDir = path.join(__dirname, '../SonoPassresources');
  const pdfFiles = fs.readdirSync(resourcesDir).filter(f => f.endsWith('.pdf'));

  console.log('Found PDF files:', pdfFiles);

  for (const pdfFile of pdfFiles) {
    console.log(`\n${'='.repeat(80)}`);
    console.log(`Processing: ${pdfFile}`);
    console.log('='.repeat(80));

    try {
      const pdfPath = path.join(resourcesDir, pdfFile);
      const text = await extractPDFText(pdfPath);

      // Save extracted text to a file for review
      const outputPath = path.join(__dirname, `${path.basename(pdfFile, '.pdf')}.txt`);
      fs.writeFileSync(outputPath, text);

      console.log(`✓ Extracted text saved to: ${outputPath}`);
      console.log(`  Text length: ${text.length} characters`);
      console.log(`  First 2000 characters:\n${text.substring(0, 2000)}`);
    } catch (error) {
      console.error(`✗ Error processing ${pdfFile}:`, error.message);
    }
  }
}

main().catch(console.error);
